files/image_config/reboot-cause/determine-reboot-cause

```diff
@@ -121,6 +126,67 @@ def find_hardware_reboot_cause():
     return hardware_reboot_cause
 
 
+def find_platform_reboot_info():
+    additional_reboot_info = None
+    # Until all platform vendors have provided sonic_platform packages,
+    # if there is no sonic_platform package installed, we don't provide
+    # additional reboot info.
+    try:
+        import sonic_platform
+
+        platform = sonic_platform.platform.Platform()
+
+        chassis = platform.get_chassis()
+
+        additional_reboot_info = chassis.get_reboot_info()
+
+    except ImportError as err:
+        log_warning("sonic_platform package not installed. Unable to retrieve platform reboot info.")
+
+    if additional_reboot_info:
+        log_info("Platform api indicates additional reboot information {}".format(additional_reboot_info))
+    else:
+        log_info("No additional reboot info from platform api")
+
+    return additional_reboot_info
+
+
+def create_prev_cause_history_file(previous_reboot_cause, comment):
+    # Create reboot_cause_#time#.json under previous_reboot_cause directory
+    FILE_GEN_TIME = str(datetime.datetime.now().strftime('%Y_%m_%d_%H_%M_%S'))
+    REBOOT_CAUSE_TIME_FILE = 'reboot-cause-' + FILE_GEN_TIME + '.json'
+    REBOOT_CAUSE_HISTORY_FILE = REBOOT_CAUSE_HISTORY_DIR + REBOOT_CAUSE_TIME_FILE
```

Comment:
Suggest renaming: `REBOOT_CAUSE_HISTORY_FILE` --> `REBOOT_CAUSE_HISTORY_FILE_PATH`

Also, suggest forming the path using `os.path.join()`